crates/types/src/identifiers.rs

```diff
@@ -479,17 +490,17 @@ impl InvocationId {
         idempotency_key: Option<&ByteString>,
     ) -> Self {
```

Comment:
There is no reason why we should accept `&ByteString` since we don't depend on how the string is allocated. Can we change this to `Option<&str>`?